src/library/scala/collection/mutable/ArrayBuilder.scala

```diff
@@ -33,14 +34,25 @@ sealed abstract class ArrayBuilder[T]
 
   protected[this] final def ensureSize(size: Int): Unit = {
     if (capacity < size || capacity == 0) {
-      var newsize = if (capacity == 0) 16 else capacity * 2
-      while (newsize < size) newsize *= 2
-      resize(newsize)
+      if (size <= 1073741824) {
+        if (size < 0) throw new Exception(
+          s"ArrayBuilder encountered negative size parameter: $size, possibly as a result of overflow."
+        ) else {
+          var newSize = if (capacity == 0) 16 else capacity * 2
+          while (newSize < size) newSize *= 2
+          resize(newSize)
+        }
+      } else if (capacity < SOFT_MAX_ARRAY_LENGTH && size <= SOFT_MAX_ARRAY_LENGTH) {
```

Comment:
As its name implies, this constant is not part of the public API of the JDK. It may not exist in all JDK implementations, which would prevent the library from building on those JDKs. For the same reasons, it does not exist in Scala.js.
We should hard code this value instead IMO.

Reply:
As a private final val, or just copy and paste the literal: `2147483639` everywhere?

Reply:
The private final val seems good to me.

Reply:
Pushed.  4th time is a charm!